### PR TITLE
Require webpack in devDependencies instead of production dependencies.

### PR DESCRIPTION
**Pull request by Krapaince**

### CODE REVIEW

**Comment by half-shell** [comment]
In order to provide a bit of context as to why this change is made, I'll just paraphrase here @Krapaince 's explanation:

---

Keeping development dependencies as normal dependencies is far from ideal. From [npm documentation](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#dependencies):
> Please do not put test harnesses or transpilers or other "development" time tools in your dependencies object

And later on:

> If someone is planning on downloading and using your module in their program, then they probably don't want or need to download and build the external test or documentation framework that you use.
In this case, it's best to map these additional items in a devDependencies object.

Which is our case here.

---

I absolutely agree to this change. I'll let @Julien29121998 weigh in if he wants to, but I'll probably merge this next week if there's no input, considering how small this change is and how reasonable it seems to me :slightly_smiling_face: 

Thanks again @Krapaince !

**Review by half-shell — APPROVED** [review]

**Comment by Julien29121998** [comment]
@Krapaince @half-shell sorry i thought i already gave my feedback for this. 
This PR makes sense and you can proceed

**Pull request merged**
